### ISSUE TITLE
[7.13.0-RC2] SystemError in LibNaCLPK.key_to_bin: C:\A\31\s\Objects\dictobject.c:1438: bad argument to internal function

### ISSUE THREAD

**Issue by kozlovsky** [Tribler/tribler#7469]
A weird bug is reported in Sentry ([GUI](https://sentry.tribler.org/organizations/tribler/issues/2179/events/bae904c9e15a46ed87aee2a68e572f2b/), [Core](https://sentry.tribler.org/organizations/tribler/issues/2180/events/596982839c2247088b4d92b41362354e/)):

```
SystemError: C:\A\31\s\Objects\dictobject.c:1438: bad argument to internal function
  File "run_tribler.py", line 106, in <module>
  File "tribler\core\start_core.py", line 205, in run_core
  File "tribler\core\start_core.py", line 170, in run_tribler_core_session
  File "asyncio\base_events.py", line 616, in run_until_complete
  File "tribler\core\start_core.py", line 120, in core_session
  File "tribler\core\components\session.py", line 58, in __aexit__
  File "tribler\core\components\session.py", line 126, in shutdown
  File "tribler\core\components\component.py", line 62, in stop
  File "tribler\core\components\ipv8\ipv8_component.py", line 141, in shutdown
  File "ipv8_service.py", line 210, in stop
  File "ipv8_service.py", line 148, in ticker
  File "ipv8_service.py", line 158, in on_tick
  File "ipv8\community.py", line 489, in get_peers
  File "ipv8\peerdiscovery\network.py", line 169, in get_peers_for_service
  File "ipv8\keyvault\public\libnaclkey.py", line 46, in key_to_bin
```

Here:
```python
class LibNaCLPK(PublicKey):
    ...
    def key_to_bin(self):
        """
        Get the string representation of this key.
        """
        return b"LibNaCLPK:" + self.key.pk + self.veri.vk    
```

It happened during the Tribler shutdown. I suspect, for some obscure reason, `self.key` or `self.veri` do not refer to a valid Python object here.

The error is reported only once, so probably not worth fixing right now. I just want to add an issue to refer to it from Sentry.

**Comment by qstokkink** [comment]
@kozlovsky do you mind if I make a copy of this issue on IPv8? Otherwise, I'd also be happy to adopt it entirely.

The error may be exotic, but this can be very easily fixed ([with a `try .. except` block around this line](https://github.com/Tribler/py-ipv8/blob/4c8db420456f209810f31ce1701cf0353180d676/ipv8_service.py#L158)).

**Comment by kozlovsky** [comment]
@qstokkink of course! Feel free to copy it to IPv8 as well!

**Comment by qstokkink** [comment]
This is fixed in [IPv8 release 2.11](https://pypi.org/project/pyipv8/2.11.0/). This issue can be closed when the IPv8 dependency version is next updated.
